src/mlpack/methods/adaboost/adaboost_impl.hpp

```diff
@@ -120,6 +120,8 @@ double AdaBoost<WeakLearnerType, MatType>::Train(
     weights = arma::sum(D);
 
     // Use the existing weak learner to train a new one with new weights.
+    // In fact, the new weak learner just inherits some hyperparameters
+    // from existing weak learner.
```

Comment:
Don't think we should talk about a new weak learning, if we remove the decision stump at some point.